tensorflow/contrib/tensor_forest/kernels/v4/fertile-stats-resource.h

```diff
@@ -64,10 +64,13 @@ class FertileStatsResource : public ResourceBase {
   // node's fertile slot's statistics if or initializes a split candidate,
   // where applicable.  Returns if the node is finished or if it's ready to
   // allocate to a fertile slot.
+  // With pre-generated feature_per_split, so it's
+  // thread safe.
   void AddExampleToStatsAndInitialize(
       const std::unique_ptr<TensorDataSet>& input_data,
       const InputTarget* target, const std::vector<int>& examples,
-      int32 node_id, bool* is_finished);
+      int32 node_id, bool* is_finished,
+      const std::vector<int>& feature_per_split = NULL);
```

Comment:
I think you wanted to provide here a const ptr instead of reference

in general the convention here is not to use the default arg values (eg remove the null and update the methods with nullptr where needed instead)